src/hipscat/pixel_tree/pixel_tree_builder.py

```diff
@@ -1,48 +1,46 @@
-from __future__ import annotations
-
 import pandas as pd
 
-from hipscat.catalog import PartitionInfo, PixelNode, PixelNodeType
-
+from hipscat.catalog import PartitionInfo
+from hipscat.pixel_tree.pixel_node import PixelNode
+from hipscat.pixel_tree.pixel_node_type import PixelNodeType
+from hipscat.pixel_tree.pixel_tree import PixelTree
 
-class PixelTree:
-    """Sparse Quadtree of HEALPix pixels that make up the HiPSCat catalog
 
-    This class stores each node in the tree, with leaf nodes corresponding to pixels with data
-    files.
+class PixelTreeBuilder:
+    """Build a PixelTree
 
-    There are a number of methods in this class which allow for quickly navigating through the
-    tree and performing operations to filter the pixels in the catalog.
+    Initially a root node is created when the builder is initialized.
+    Nodes can then be added to the tree.
+    To create a pixel tree object once the tree is built, call the `build` method
 
-    Attributes:
-        pixels: Nested dictionary of pixel nodes stored in the tree. Indexed by HEALPix
-        order then pixel number root_pixel: Root node of the tree. Its children are a subset of the
-        12 base HEALPix pixels
     """
 
-    def __init__(self, partition_info_df: pd.DataFrame):
-        """Initialises the Tree from the partition info metadata
+    def __init__(self):
+        self.root_pixel = PixelNode(-1, -1, PixelNodeType.ROOT, None)
+        self.pixels = {-1: {-1: self.root_pixel}}
 
-        Args:
-            partition_info_df: Partition Info metadata file loaded into a pandas DataFrame
+    def build(self) -> PixelTree:
+        """Build a `PixelTree` object from the nodes created
 
-        Raises:
-            ValueError: Incorrectly configured pixel structure in metadata file
+        Returns:
+            The pixel tree with the nodes created in the builder
         """
-        self.root_pixel = PixelNode(-1, -1, PixelNodeType.ROOT, None)
-        self.pixels = {-1: {-1: self.root_pixel}}
-        self._create_tree(partition_info_df)
+        return PixelTree(self.root_pixel, self.pixels)
 
-    def __len__(self):
-        """Gets the number of nodes in the tree, including inner nodes and the root node
+    @staticmethod
+    def from_partition_info_df(partition_info_df: pd.DataFrame) -> PixelTree:
+        """Build a tree from ta partition info DataFrame
```

Comment:
nit: is "ta" an acronym here?

Reply:
nope, that's a typo, thanks for spotting that!